### PR TITLE
Standardize Imports I, #176

### PR DESCRIPTION
**Pull request by zuphilip**
The import statements for the main commands ocropus-* are
standardized:
 * Put different import statements into different lines
 * Group them into (1) standard imports, (2) third party
   imports, (3) local modules
 * Avoid the "from pylab import *"
 * Use rather "import numpy as np", "import matplotlib.pyplot as plt"

Moreover, some unused import statements were deleted and a very
few nits for readibility are done.

This work is heavily based on the work by @QuLogic in the branch
https://github.com/QuLogic/ocropy/commits/standard-import
See also #176 .

All commands from the current pipeline are tested to still work
in the usual calls (especially run-test and run-rtrain). Even more, the
outputs before and afterwards is exactly the same.

### CODE REVIEW

**Comment by zuphilip** [comment]
I tried to automatically check the addition of a namespace to the relevant functions (i.e. these are the ones I found so far in the ocropus code) with these lines:
```
# used functions from numpy need now a prefix 'np.'
perl -ne 'print "$. $_" if /([^.]|^)(amax|amin|arange|array|asarray|clip|dtype\(|isnan|linspace|maximum\(|mean|median\(|minimum\(|ones|random|rand|randn|randint|transpose|var|vstack|zeros)/' ocropus-*

# used functions from matplotlib.pyplot need now a prefix 'plt.'
perl -ne 'print "$. $_" if /([^.]|^)(clf|cm.gray|cm.hot|draw|figure|gca|ginput|gray\(|imshow|\bion\(|\bplot|rc\(|rcParams|savefig|subplot|title)/' ocropus-*
```

**Comment by ChillarAnand** [comment]
Looks good. Any updates on this? We can also use importmagic to automatically fix imports.

File: ocropus-dewarp [hunk 1]
```diff
@@ -47,8 +43,8 @@ for fname in inputs:
     base,_ = ocrolib.allsplitext(fname)
     try:
         line = ocrolib.read_image_gray(fname)
-        lnorm.measure(amax(line)-line)
-        line = lnorm.normalize(line,cval=amax(line))
+        lnorm.measure(np.amax(line)-line)
+        line = lnorm.normalize(line,cval=np.amax(line))
         scipy.misc.imsave(base+".dew.png",line)
```
Comment (zuphilip) [inline ocropus-dewarp]: We still need `scipy` here, cf. Travis output.

File: ocropus-linegen [hunk 2]
```diff
@@ -114,36 +122,41 @@ assert len(lines)>0
...
 
+
 def rgeometry(image,eps=0.03,delta=0.3):
-    m = array([[1+eps*randn(),0.0],[eps*randn(),1.0+eps*randn()]])
+    m = np.array([[1+eps*np.random.randn(),0.0],[eps*np.random.randn(),1.0+eps*np.random.randn()]])
     w,h = image.shape
-    c = array([w/2.0,h/2])
-    d = c-dot(m,c)+array([randn()*delta,randn()*delta])
+    c = np.array([w/2.0,h/2])
+    d = c-dot(m,c)+np.array([np.random.randn()*delta,np.random.randn()*delta])
```
Comment (zuphilip) [inline ocropus-linegen]: `NameError: global name 'dot' is not defined` -> Thus, I guess this should be `np.dot(...)`

**Comment by zuphilip** [comment]
@ChillarAnand Any checking with `importmagic` would be appreciated. Moreover, I would like to wait for #234.

**Comment by ChillarAnand** [comment]
importmagic doesnt work out of the box. I have written a wrapper script and it works only on Python 3. 

Any timeline for [this PR](https://github.com/tmbdev/ocropy/pull/114)?

**Comment by zuphilip** [comment]
Okay, I understand. Actually, we still concentrate on Python 2 and the PR you mentioned is for a separate Python 3 branch. In general it might be hard work to have the whole OCRopus work for Python 2 and Python 3.

**Comment by zuphilip** [comment]
Wow, that was fast :-) I hope that no error was created in the last round of solving merge conflict, because I didn't test it myself much, but simply relied on Travis and the extended tests we have now...

**Comment by kba** [comment]
:shipit: :shipit: :shipit: 

But seriously, if we'll see regressions from this, it will be fairly obvious, missing imports or undeclared vars.

**Pull request merged**